tools/setup_helpers/cmake.py

```diff
@@ -115,13 +115,17 @@ def _get_cmake_command():
         if IS_WINDOWS:
             return cmake_command
         cmake3 = which('cmake3')
+        cmake = which('cmake')
         if cmake3 is not None:
-            cmake = which('cmake')
             if cmake is not None:
                 bare_version = CMake._get_version(cmake)
                 if (bare_version < LooseVersion("3.5.0") and
                         CMake._get_version(cmake3) > bare_version):
                     cmake_command = 'cmake3'
+            else:
+                cmake_command = 'cmake3'
+        elif cmake is None:
+            raise RuntimeError('no cmake or cmake3 found')
         return cmake_command
```

Comment:
I can foresee some issues. If only one of `cmake` and `cmake3` is found, and it is prior to 3.5.0, things will go busted. Actually I think it would be way easier to rewrite this whole function to make the logic clearer.

Reply:
Thanks for the advice! I will rewrite the function. 
A further question, do versions of cmake and cmakes must  be greater than 3.5.0?
The original code don't check the version of cmake when cmake3 doesn't exist.

Reply:
Yes, they must be: https://github.com/pytorch/pytorch/blob/bcb44796ba00f9ac5f22e33f1de3a24b277c4c3a/CMakeLists.txt#L1

Reply:
Much appreciated!
I have rewriten the function to make it clear.